public/app/features/explore/AddToDashboard/addToDashboard.ts

```diff
@@ -0,0 +1,44 @@
+import { DataQuery } from '@grafana/data';
+import { DashboardDataDTO } from 'app/types';
+import { getDashboardSrv } from 'app/features/dashboard/services/DashboardSrv';
+import { getBackendSrv } from '@grafana/runtime';
+import { lastValueFrom } from 'rxjs';
+
+export interface SaveToNewDashboardDTO {
+  dashboardName: string;
+  folder: { title: string; id: number };
+  queries: DataQuery[];
+  visualization: string;
+}
+
+export interface SaveToExistingDashboardDTO {
+  dashboard: string;
+  queries: DataQuery[];
+  visualization: string;
+}
+
+const createDashboardApiCall = (dashboard: DashboardDataDTO, folderId: number) => {
```

Comment:
still WIP, i'd like to move this to `dashboardSrv`, need to check with user essentials